docs/kubernetes_migration_guide.md

```diff
@@ -0,0 +1,43 @@
+---
+title: Kubernetes Migration Guide
+layout: docs
+---
+
+_Document outline_
+
+## Migrating to Kubernetes
+- Are you migrating to Kubernetes?
+- This guide is for you.
+- Tilt helps developments teams do development, that are already deploying to Kubernetes now, or are planning to do so in the future.
+- Many teams focus on the production aspect of Kubernetes. They often don't realize that it has downstream impacts on development. This guides puts both together in perpsective.
+
+## Production migration
+- Here are some resources on production migration of Kubernetes.
+- Tilt is not designed for deploying to production.
+
+## Development migration
+- We've seen many teams migrate to Kubernetes for production, without much thought of the impacts on development.
+- In the worst case, development teams have needed to scramble to adjust existing development workflows to accommodate Kubernetes in production, on strict timelines.
+- We recommend that you have a development migration plan in conjunction with your production migration plan.
+- It's a best practice to decouple (as much as possible) the dev plan and the prod plan. Lots of moving pieces and inherent risk to get things working altogether at once.
+- If you are the dev team, there's no need to wait for production migration. You can do development migration first, in short:
```

Comment:
This says "use Tilt for dev on Kubernetes before using prod on Kubernetes".

I think the more important message is "use Tilt for dev on non-Kubernetes before doing dev on Kubernetes. This makes it easier to switch dev to Kubernetes in the future."

(This is only relevant to people who have non-Kubernetes dev already).

Reply:
Ah this is interesting @dbentley . Tilt can be used without k8s. (But needs at least Docker right? I.e. replacing docker-compose? https://docs.tilt.dev/docker_compose.html). So the migration path would be:

- Team is using Docker containers and docker-compose
- Team uses Docker containers and Tilt
- Team uses Docker containers and local Kubernetes cluster and Tilt
- Production deploy uses Kubernetes

Reply:
You can use Tilt with bare processes with local_resource. https://blog.tilt.dev/2020/02/12/local-dev.html

Reply:
Oh great! There's so much that can be done in terms of guides / example projects here. Will make sure to get Nick's take before more revisions here.